File: android/pi-ai/src/main/kotlin/com/pimobile/ai/ProtocolAdapter.kt
package com.pimobile.ai

import kotlinx.coroutines.flow.Flow
import okhttp3.Request
import okhttp3.Response

interface ProtocolAdapter {
    fun buildRequest(context: Context, model: ModelDefinition, apiKey: String): Request
    fun parseStreamEvents(response: Response): Flow<StreamEvent>
}

File: ios/PiAI/Sources/PiAI/ProtocolAdapter.swift
import Foundation

/// Adapters transform between our generic types and provider-specific HTTP requests/responses.
public protocol ProtocolAdapter: Sendable {
    /// Build a URLRequest for the given context and model.
    func buildRequest(context: Context, model: ModelDefinition, apiKey: String) throws -> URLRequest

    /// Parse SSE stream lines into StreamEvents.
    func parseStreamEvents(lines: AsyncLineSequence<URLSession.AsyncBytes>) -> AsyncThrowingStream<StreamEvent, Error>
}
